### PR TITLE
CUMULUS-3238: Remove ElasticSearch dependency from Collection POST, PUT, and DEL endpoints

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: packages/api/endpoints/collections.js [hunk 1]
```diff
@@ -238,44 +223,26 @@ async function del(req, res) {
...
     if (error instanceof RecordDoesNotExist) {
-      if (!(await esCollectionsClient.exists(collectionId))) {
-        log.info('Collection does not exist in Elasticsearch and PostgreSQL');
-        return res.boom.notFound('No record found');
-      }
-      log.info('Collection does not exist in PostgreSQL, it only exists in Elasticsearch. Proceeding with deletion');
-    } else {
-      throw error;
+      log.info(`Collection does not exist in PostgreSQL. Failed to delete collection with name ${name} and version ${version}`);
+      return res.boom.notFound('No record found');
```
Comment (npauzenga) [inline packages/api/endpoints/collections.js:239]: I think we still need the `else throw error`, right? If the error is not RecordDoesNotExist, I believe this will swallow the error unless there's a catch further up I'm not looking at (which is possible).

File: packages/api/tests/endpoints/collections/delete-collection.js [hunk 2]
```diff
@@ -386,112 +305,6 @@ test.serial('Attempting to delete a collection with an associated rule does not
   ));
 });
 
-test.serial('del() does not remove from Elasticsearch or publish SNS message if removing from PostgreSQL fails', async (t) => {
```
Comment (npauzenga) [inline packages/api/tests/endpoints/collections/delete-collection.js:389]: This test might still be valid without ES? e.g. `del() does not publish SNS message if removing from PostgreSQL fails`

File: packages/api/tests/endpoints/collections/update-collection.js [hunk 3]
```diff
@@ -435,12 +352,11 @@ test.serial('PUT returns 400 for version mismatch between params and payload', a
   t.falsy(record);
 });
 
-test.serial('PUT does not write to Elasticsearch or publish SNS message if writing to PostgreSQL fails', async (t) => {
+test.serial('PUT does not write publish SNS message if writing to PostgreSQL fails', async (t) => {
```
Comment (npauzenga) [inline packages/api/tests/endpoints/collections/update-collection.js:355]:
```suggestion
test.serial('PUT does not publish SNS message if writing to PostgreSQL fails', async (t) => {
```

**Review by npauzenga — CHANGES_REQUESTED** [review]
This looks great @Nnaga1! Just getting into the ES removal stuff so this is helpful.

File: packages/api/endpoints/collections.js (hunk 1 above)
Comment (npauzenga) [reply packages/api/endpoints/collections.js:239]: All good. We can mark things as WIP and/or put them in Draft status in the future but not a big deal.

File: packages/api/endpoints/collections.js [hunk 4]
```diff
@@ -21,11 +20,6 @@ const {
   CollectionSearch,
 } = require('@cumulus/db');
 const CollectionConfigStore = require('@cumulus/collection-config-store');
-const { getEsClient, Search } = require('@cumulus/es-client/search');
-const {
-  indexCollection,
```
Comment (etcart) [inline packages/api/endpoints/collections.js:26]: I want to be clear that the intention is to leave this function (indexCollection) in the code base?
I see that it's used in the api to provide ES interfaces, is the plan to leave the api still able to interact with ES so people can manage their cluster even while we're not adding new data to it?

File: packages/api/endpoints/collections.js [hunk 5]
```diff
@@ -176,7 +166,6 @@ async function put(req, res) {
   const {
     collectionPgModel = new CollectionPgModel(),
     knex = await getKnexClient(),
-    esClient = await getEsClient(),
     collectionConfigStore = new CollectionConfigStore(
       process.env.system_bucket,
```
Comment (etcart) [inline packages/api/endpoints/collections.js:170]: this is slight scope creep so *nit* but
getRequiredEnvVar('system_bucket', process.env) here, and elsewhere in this file
(107, 108, 171, 172, 228, 229) would shift us towards better typing while you're in this file

File: packages/api/tests/endpoints/collections/delete-collection.js [hunk 6]
```diff
@@ -161,6 +145,54 @@ test('Attempting to delete a collection without an Authorization header returns
   );
 });
 
+test.serial('del() does not publish a SNS message if removing from PostgreSQL fails', async (t) => {
```
Comment (etcart) [inline packages/api/tests/endpoints/collections/delete-collection.js:148]:
```suggestion
test.serial('del() does not publish an SNS message if removing from PostgreSQL fails', async (t) => {
```

File: packages/api/tests/endpoints/collections/update-collection.js [hunk 7]
```diff
@@ -270,49 +236,7 @@ test.serial('PUT replaces an existing collection and correctly removes fields',
   });
 });
 
-test.serial('PUT replaces an existing collection in PG with correct timestamps', async (t) => {
```
Comment (etcart) [inline packages/api/tests/endpoints/collections/update-collection.js:273]: why was this cut out?

**Review by etcart — CHANGES_REQUESTED** [review]
mostly this is great, I'd like to clarify a few things to be sure they are what's intended

File: packages/api/endpoints/collections.js (hunk 5 above)
Comment (etcart) [reply packages/api/endpoints/collections.js:170]: yes exactly

**Review by etcart — APPROVED** [review]

**Review by etcart — APPROVED** [review]
this all looks good, ready to go.
my concerns and Nates have both been addressed and I don't see anything else that raises concerns
